### PR TITLE
Generating unique ID for each picture

### PR DESCRIPTION
**Pull request by mvshao**
**Description**
Generating unique ID and handling 404 errors. 

**Changes:**
- nanoID generating id for each entry
- handling EOF error for JSON
- CORS policy for fetching data
- 404 error handling
- updated data model for entries in the database
- updated unit tests

Related issue #14

### CODE REVIEW

File: backend/database_test.go [hunk 1]
```diff
@@ -218,7 +217,7 @@ func TestGetAllKeysDB(t *testing.T) {
 			t.Error(err)
 		}
 	})
-	t.Run("should get all keys from database with no errors", func(t *testing.T) {
+	t.Run("should no error when all keys have been got from database", func(t *testing.T) {
```
Comment (mjakobczyk) [inline backend/database_test.go:220]: Rather: `should return no error (...)`

File: backend/model.go [hunk 2]
```diff
@@ -2,7 +2,12 @@ package main
...
 type Image struct {
-	URL string `json:"url"`
-	GCP string `json:"gcp"`
-	IMG string `json:"img"`
+	ID      string `json:"id"`      //nanoID
+	Content string `json:"content"` //base64
+	GCP     string `json:"gcp"`     //returned JSON from gcp api
+	Status  bool   `json:"status"`
+}
+
```
Comment (mjakobczyk) [inline backend/model.go:10]: No docs.

File: backend/utils.go [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package main
+
+import "github.com/matoous/go-nanoid/v2"
+
+type idGenerator struct{}
+
+func NewIdGenerator() *idGenerator {
```
Comment (mjakobczyk) [inline backend/utils.go:7]: No docs.

File: backend/utils.go [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+	return &idGenerator{}
+}
+
```
Comment (mjakobczyk) [inline backend/utils.go:10]: No docs.

File: backend/utils.go [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+func (g *idGenerator) NewID() (string, error) {
+	return gonanoid.New()
+}
+
+// TODO: Finish documentation
```
Comment (mjakobczyk) [inline backend/utils.go:15]: Finish this TODO please.

**Review by mjakobczyk — APPROVED** [review]

**Review by Lyczeq — APPROVED** [review]
/lgtm

**Pull request merged**